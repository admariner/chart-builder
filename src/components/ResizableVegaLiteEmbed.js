// @flow
/* eslint no-unused-vars: ["warn", { "args": "after-used" }] */
import React, { Component } from 'react'
import { DraggableCore } from 'react-draggable'
import VegaLiteEmbed from './VegaLiteEmbed'
import classes from './ResizableVegaLiteEmbed.module.css'

type Props = {
  spec: Object,
  data: Array<Object>,
  onViewRender: (v: Object) => void,
  showResize: boolean,
  setDimensions: (width: number, height: number) => mixed
}

export default class ResizableVegaLiteEmbed extends Component<Props> {
  handleDrag = (e: any, dragData: Object) => {
    this.props.setDimensions(dragData.x, dragData.y)
  }

  render() {
    return (
      <div className={classes.container}>
        <VegaLiteEmbed {...this.props} key={JSON.stringify(this.props.spec)} />
        {this.props.showResize && (
          <DraggableCore onDrag={this.handleDrag}>
            <span className={classes.handle} />
          </DraggableCore>
        )}
      </div>
    )
  }
}
