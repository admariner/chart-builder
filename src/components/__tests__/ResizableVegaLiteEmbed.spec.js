import React from 'react'
import ResizableVegaLiteEmbed from '../ResizableVegaLiteEmbed'

jest.mock('react-draggable', () => ({ DraggableCore: 'draggable-core' }))
jest.mock('../VegaLiteEmbed', () => 'vega-lite-embed')

it('renders', () => {
  snap(
    <ResizableVegaLiteEmbed
      spec={{ testspec: 1 }}
      data={[{ data: 1 }]}
      onViewRender={() => {}}
      showResize
      setDimensions={() => {}}
    />
  )
})

it('renders without resize', () => {
  snap(
    <ResizableVegaLiteEmbed
      spec={{ testspec: 1 }}
      data={[{ data: 1 }]}
      onViewRender={() => {}}
      showResize={false}
      setDimensions={() => {}}
    />
  )
})
