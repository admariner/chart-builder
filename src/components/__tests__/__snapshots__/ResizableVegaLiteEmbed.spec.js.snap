// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div
  className="container"
>
  <vega-lite-embed
    data={
      Array [
        Object {
          "data": 1,
        },
      ]
    }
    onViewRender={[Function]}
    setDimensions={[Function]}
    showResize={true}
    spec={
      Object {
        "testspec": 1,
      }
    }
  />
  <draggable-core
    onDrag={[Function]}
  >
    <span
      className="handle"
    />
  </draggable-core>
</div>
`;

exports[`renders without resize 1`] = `
<div
  className="container"
>
  <vega-lite-embed
    data={
      Array [
        Object {
          "data": 1,
        },
      ]
    }
    onViewRender={[Function]}
    setDimensions={[Function]}
    showResize={false}
    spec={
      Object {
        "testspec": 1,
      }
    }
  />
</div>
`;
